File: FinBoard/BackEnd/src/main/java/pwork/greco/antonio/finboard/entity/Portfolio.java
package pwork.greco.antonio.finboard.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "Portfolio")
public class Portfolio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "UserId")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CompanyId")
    private Company company;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private Boolean deleted;
}

File: FinBoard/FrontEnd/src/app/models/Portfolio.ts
export interface Portfolio {
  id: number;                   // bigint
  createdAt: string;            // datetime2(6)
  deleted: boolean;             // bit
  name: string;                 // varchar(50)
  companyId?: number | null;    // bigint nullable (FK a company)
  userId?: number | null;       // bigint nullable (FK a user)
}
